nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-runtime/src/main/java/org/apache/nifi/NiFi.java

```diff
@@ -305,39 +308,73 @@ private static String loadFormattedKey(String[] args) {
         String key = null;
         List<String> parsedArgs = parseArgs(args);
         // Check if args contain protection key
-        if (parsedArgs.contains(KEY_FLAG)) {
-            key = getKeyFromArgs(parsedArgs);
-
+        if (parsedArgs.contains(KEY_FILE_FLAG)) {
+            key = getKeyFromKeyFileAndPrune(parsedArgs);
             // Format the key (check hex validity and remove spaces)
             key = formatHexKey(key);
-            if (!isHexKeyValid(key)) {
-                throw new IllegalArgumentException("The key was not provided in valid hex format and of the correct length");
-            }
 
-            return key;
-        } else {
-            // throw new IllegalStateException("No key provided from bootstrap");
+        }
+
+        if (null == key) {
             return "";
+        } else if (!isHexKeyValid(key)) {
+          throw new IllegalArgumentException("The key \""+key+"\" was not provided in valid hex format and of the correct length");
```

Comment:
+1. It's also the non-expected case; the key would normally be generated by encrypt-config, which wouldn't generate invalid keys.